airbyte-commons-worker/src/test/java/io/airbyte/workers/RecordSchemaValidatorTest.java

```diff
@@ -26,28 +28,57 @@ class RecordSchemaValidatorTest {
   private static final String FIELD_NAME = "favorite_color";
   private static final AirbyteMessage VALID_RECORD = AirbyteMessageUtils.createRecordMessage(STREAM_NAME, FIELD_NAME, "blue");
   private static final AirbyteMessage INVALID_RECORD = AirbyteMessageUtils.createRecordMessage(STREAM_NAME, FIELD_NAME, 3);
+  private ConcurrentHashMap<AirbyteStreamNameNamespacePair, ImmutablePair<Set<String>, Integer>> validationErrors;
 
   @BeforeEach
   void setup() throws Exception {
     final ImmutablePair<StandardSync, StandardSyncInput> syncPair = TestConfigHelpers.createSyncConfig();
     syncInput = syncPair.getValue();
+    validationErrors = new ConcurrentHashMap<>();
   }
 
   @Test
-  void testValidateValidSchema() throws Exception {
-    final var featureFlagClient = new TestClient();
-    final var recordSchemaValidator = new RecordSchemaValidator(featureFlagClient, syncInput.getWorkspaceId(),
-        WorkerUtils.mapStreamNamesToSchemas(syncInput));
-    recordSchemaValidator.validateSchema(VALID_RECORD.getRecord(), AirbyteStreamNameNamespacePair.fromRecordMessage(VALID_RECORD.getRecord()));
+  void testValidateValidSchema() {
+    final var recordSchemaValidator = new RecordSchemaValidator(WorkerUtils.mapStreamNamesToSchemas(syncInput), false);
+    for (var i = 0; i < 10; i++) {
```

Comment:
nit: would be clearer if `i` were named something to reflect the validation threshold.

Reply:
curious, why do 10 times? If the record value isn't changing each time, seems not necessary to check 10 times.